docs/contributing/Git-mini-guide.md

```diff
@@ -0,0 +1,196 @@
+You may use SSH keys. You may use PAT.
+See the Git-book and
+https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls
+
+
+Now you've signed in, create a fork.
+  Browse to https://github.com/wxWidgets/wxWidgets
+  About the top right corner find and use the "Fork" button. Select "Create a new fork"
+The new fork will be created as github.com/<your_github_name>/wxWidgets.git
+
+This fork is your fork. You don't need another fork to submit changes for
+another contribution to wxWidgets.
+
+4. Download wxWidgets code, now yes
+-----------------------------------
+Create the directory 'wxFork' (or the name you like).
+
+Option a)
+  git clone --recurse-submodules https://github.com/wxWidgets/wxWidgets.git wxFork
+Option b)
+  git clone --recurse-submodules https://github.com/<your_github_name>/wxWidgets.git wxFork
+
+This will populate the 'wxFork' directory in your local PC with all last
+("master branch") wx sources.
+
+'clone' stores the remote (github.com/xxxxx) and labels it as "origin".
+The active branch (see '5.') in that remote is labeled as "master".
+
+The difference between a) and b) is met when you want to "push" your changes
+to wxWidgets repository. See point '7.' some lines below.
+
+If you need to update your local copy with the current "remote" wxWidgets
+master, you can use:
+  git pull --ff-only
+
+The files downloaded with this command will be stored for your local active "branch"
+
+5. Branches. Work your changes locally
+--------------------------------------
+Same as you need a 'fork' in GitHub you want a 'branch' (read: divergence) in you local PC.
+  git checkout -b mybranch origin/master
+
+This will create (the '-b' flag) a branch named "mybranch", which reflects all
+changes since the starting point 'origin/master' (what you did at step 4).
+'checkout' will also "jump" (read: set as active) to that branch.
+
+You may have several branches (say you work for different patches).
+You can jump to an existing branch named 'br2' by
+  git checkout br2
+To list all available branches use
+  git branch
+To delete the branch "badbranch" use
+  git branch -d badbranch
+
+An important feature of Git is that if you open a file with your editor you will see
+the version that exists in the active branch. If you checkout other branch, you'll
+see other version (if it's different, obviously).
+In other words, Git changes your working directory. So be sure to switch to the
+proper branch before any work.
+
+Make your changes, compile and test them in that branch you jumped to. Only one
+branch per contribution, please.
+
+6. Prepare for sharing
+-----------------------
+Not all changed files will be part of your contribution. No compiled files,
+just code files.
+
+  git status -v
+Inspect the output of this command. It will show the files that you changed
+and/or added.
+
+Tell Git the staged files:
+  git add --all

```

Comment:
Ironically, this PR itself has proven using `add --all` is a bad idea.

Reply:
No, I didn't use 'git add --all' but added the file.
The issue with this PR (mixing PRs) is that when I created the new branch in my local I forgot to checkout to master first, so the new branch hangs from the other branch instead of the master one.